main.go

```diff
@@ -78,20 +80,33 @@ func main() {
 		LeaderElectionID:       "ae40cfcb.openshift.io",
 	})
 	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+		setupLog.Error(err, "unable Start new manager")
+
+		return mgr, fmt.Errorf("starting new manager failed %w", err)
+	}
+
+	return mgr, nil
+}
+
+func main() {
+	mgr, err := newManager()
```

Comment:
Ack

> We need to get some testing going. PR is fine (just some nits mostly).
> 
> Once PR #8 is in, we can leverage the testing framework to add required tests on PVC events? (or what is the plan :) )

Agree. I thought about adding tests. The reason is that, the pvc reconcile requests are coupled with creation of VolumeReplication CR. So, I thought of deferring the addition of tests until those changes also come in. But if you think that pvc detection in itself should have its own unit tests, then I will add the tests.

Reply:
> (nit) follow `if err = Function(); err != nil` style where possible

Ack

Reply:
> (nit) follow `if err = Function(); err != nil` style where possible

I checked this and here is my observation. Atleast in this context (i.e. for this part of the code in the main function from main.go), it would be better to keep it as it is (i.e. dont do function calling and error checking in one line).

The reason being, with such one liners, the scope of "mgr" and "err" would be limited to that "if" block only. The code below it (i.e. the code after the "if" block) which would refer both "err" and "mgr" variables would error out saying they are not defined. 

Not having the one liner check would define the variables "mgr" and "err" whose scope would be for the entire function which would be useful.

Please let me know if I am missing something.

Reply:
> Please remember the rule, this is a nit and does not stop this PR from merging. IOW, it is fine as it stands.

ACK. Sure. :)